cartography/intel/trivy/scanner.py

```diff
@@ -361,3 +374,61 @@ def sync_single_image_from_s3(
             f"Failed to process S3 scan results for {image_uri} from {s3_object_key}: {e}"
         )
         raise
+
+
+@timeit
+def read_scan_results_from_file(
+    file_path: str,
+) -> tuple[str | None, list[dict], str | None]:
+    """Read and parse Trivy scan results from a local file."""
+    logger.debug(f"Reading scan results from file: {file_path}")
+    with open(file_path, encoding="utf-8") as f:
+        trivy_data = json.load(f)
+
+    artifact_name = trivy_data.get("ArtifactName")
+
+    if "Results" in trivy_data and trivy_data["Results"]:
+        results = trivy_data["Results"]
+    else:
+        stat_handler.incr("image_scan_no_results_count")
+        logger.warning(
+            f"File scan data did not contain a `Results` key for {file_path}; continuing."
+        )
+        results = []
+
+    image_digest = None
+    if "Metadata" in trivy_data and trivy_data["Metadata"]:
+        repo_digests = trivy_data["Metadata"].get("RepoDigests", [])
+        if repo_digests:
+            repo_digest = repo_digests[0]
+            if "@" in repo_digest:
+                image_digest = repo_digest.split("@")[1]
```

Comment:
If having an image digest is critical to other things later on in the workstream, the let's make it not optional and have it raise exc here if not present.

Otherwise if not critical, we can leave it it as it is.

Reply:
done, digests are critical